### PR TITLE
fixes bug 1313676 - Optimize how crashes are cloned in processor

### PR DESCRIPTION
**Pull request by peterbe**
**tl;dr here's the executive summary of my measurements...**
```
____OLD_WAY____
# uuids 31
# measurements 93
time per uuid 0.252148882035
____NEW_WAY____
# uuids 31
# measurements 124
time per uuid 0.170331147409

average time just to make the pure dict copy 0.0462440675305
```

I took 31 random uuids, instrumented some `t0=time.time(); do_something(); t1=time.time()` in the master branch. Ran a bunch of processing and dumped all the measurements to a file. 
Then I wrote this new code, and put time measurements again. This time I had to do one measurement for making the pure dict first and 3 measurements for each cloning. 

See, 31 uuids --> 3x31=93 measurements. And with the new way, 31+3x31=124. 

It's promising. This small sample indicates that it'll be ~30% faster this new way just for this "region" of the code. In other words, processing won't become 30% faster (if someone else reads this and gets too excited)

### CODE REVIEW

**Comment by peterbe** [comment]
cc @adngdb

File: socorro/external/crashstorage_base.py [hunk 1]
```diff
@@ -612,27 +612,32 @@ def save_raw_and_processed(self, raw_crash, dump, processed_crash,
...
+            on a SocorroDotDict object so the crash in this context needs
+            to be pure python dict.
 
             Ideally, we shouldn't be using any of this *DotDict and just
             use plain dict objects.
             """
-            return SocorroDotDict(
-                copy.deepcopy(socorrodotdict_to_dict(crash))
-            )
+            return SocorroDotDict(copy.deepcopy(crash))
```
Comment (peterbe) [inline socorro/external/crashstorage_base.py]: good point. will do.

**Comment by peterbe** [comment]
@willkg Inlined now and rewritten the code comment.

**Comment by willkg** [comment]
Looks good to me. r+
